tests/yam/agama/boot_agama.pm

```diff
@@ -96,16 +96,19 @@ sub run {
     my @params = prepare_boot_params();
 
     $grub_menu->expect_is_shown();
-    $grub_menu->select_check_installation_medium_entry() if get_var('AGAMA_GRUB_CHECK_MEDIUM');
+    $grub_menu->select_check_installation_medium_entry() if get_var('AGAMA_GRUB_SELECTION') =~ /check_medium/;
+    $grub_menu->select_rescue_system_entry() if get_var('AGAMA_GRUB_SELECTION') =~ /rescue_system/;
```

Comment:
as this setting is not going to contain a list, better go with `check_var` instead of regex.